apps/flame-defi/app/features/evm-wallet/services/generic-contract-service.ts

```diff
@@ -35,11 +61,9 @@ export class GenericContractService {
     methodName: string,
     args: unknown[] = [],
     value?: bigint,
+    gasLimit?: bigint,
   ): Promise<Hash> {
-    const walletClient = await getWalletClient(this.wagmiConfig, { chainId });
-    if (!walletClient) {
-      throw new Error("No wallet client available");
-    }
+    const walletClient = await this.getWalletClient(chainId);
```

Comment:
I think I had this in the slightly wrong spot previously. Let's move this to inside the try/catch below.